### ISSUE TITLE
[Eslint] Add rule for consistent tabIndex formatting

### ISSUE THREAD

**Issue by macandcheese** [Esri/calcite-design-system#7631]
### Summary

Across the library we currently have a lot of `tabindex` and `tabIndex` used. 

### Desired Outcome

Add a linting rule for correct `tabIndex` casing.

### Resources

cc @driskull

**Comment by jcfranco** [comment]
Blocked by #10310.

**Comment by jcfranco** [comment]
Lumina enforces prop-usage, so this is no longer needed. 🎉 Closing!
